src/daemon/ca/keys.rs

```diff
@@ -124,22 +124,20 @@ impl CertifiedKey {
                 new_not_after.to_rfc3339()
             );
             false
-        } else if until_not_after_millis == until_new_not_after_millis {
+        } else if until_not_after == until_new_not_after {
             debug!(
                 "Will not request new certificate for CA '{}' under RC '{}'. Resources and not after time are unchanged.",
                 handle,
                 rcn,
             );
             false
-        } else if until_new_not_after_millis < until_not_after_millis {
+        } else if until_not_after > 0 && (until_new_not_after as f64 / until_not_after as f64) < 0.9_f64 {
```

Comment:
With these comparisons, as the timestamp value gets further into the future the same number of difference in seconds between `until_not_after` and `until_new_not_after` will change as a percentage, e.g. 200/240=0.83 but 400/440=0.90. If we take the unix epoch timestamp in seconds now as being 1645015173 then even roughly a month of difference won't be 10% in epoch time: (1645015173 - 86400*30) / 1645015173 = 0.998 which is not `< 0.9_f64`. I'm not sure if I've understood this correctly. Is this what is wanted?

Reply:
The rename to "remaining" time made it clear that the numbers compared are not absolute unix epoch timestamps but seconds remaining and thus do not lead to huge variable differences and thus it is now clear that this does what was intended.

One thought: is this 10% behaviour covered somewhere by test cases?